lib/routes/client/register.js

```diff
@@ -41,6 +42,8 @@ module.exports = {
   handler: function registerEndpoint(req, reply) {
     var payload = req.payload;
     var secret = unique.secret();
+    var authHeader = req.headers.authorization;
```

Comment:
You can access this via hapi's `auth` property. `req.auth.credentials` will be an object like this: https://github.com/mozilla/fxa-oauth-server/blob/master/lib/token.js#L18